src/main/java/com/igot/cb/metrics/service/ApiMetricsTracker.java

```diff
@@ -0,0 +1,247 @@
+package com.igot.cb.metrics.service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ApiMetricsTracker {
+
+    // Centralized map to store metrics for each API endpoint
+    private static final Map<String, ApiMetrics> apiMetricsMap = new ConcurrentHashMap<>();
+    private static final ThreadLocal<Boolean> trackingEnabled = ThreadLocal.withInitial(() -> false);
+
+    public static void enableTracking() {
+        trackingEnabled.set(true);
+    }
+
+    /**
+     * Records an API call.
+     *
+     * @param apiEndpoint The API endpoint being tracked.
+     */
+    public static void recordApiCall(String apiEndpoint) {
+        if (!trackingEnabled.get()) {
+            return;
+        }
+        ApiMetrics apiMetrics = apiMetricsMap.computeIfAbsent(apiEndpoint, k -> new ApiMetrics());
+        apiMetrics.incrementApiCallCount();
+    }
+
+    /**
+     * Records a database operation's metrics.
+     *
+     * @param apiEndpoint The API endpoint being tracked.
+     * @param dbName      The name of the database (e.g., "cassandra", "redis").
+     * @param operation   The operation performed (e.g., "insert", "read").
+     * @param timeTaken   The time taken for the operation in milliseconds.
+     */
+    public static void recordDbOperation(String apiEndpoint, String dbName, String operation, long timeTaken) {
+        if (!trackingEnabled.get()) {
+            return;
+        }
```

Comment:
this has to be other way around... you need to check this from caller before calling recordDbOperation